File: merge_sort/mergeSort.py
def mergeSort(nums):
    length = len(nums)
    if length < 2:
        return nums

    middle = length//2
    first = nums[:middle]
    last = nums[middle:]

    return stitch(mergeSort(first), mergeSort(last))


def stitch(first, last):
    results = []
    while len(first) > 0 and len(last) > 0:
        if first[0] < last[0]:
            results.append(first.pop(0))
        else:
            results.append(last.pop(0))

    return results + first + last

File: merge_sort/mergeSort.js
const mergeSort = (nums) => {
  const length = nums.length;
  if (length < 2) {
    return nums;
  }
  let mid = Math.floor(length / 2);
  let start = nums.slice(0, mid);
  let last = nums.slice(mid);

  return stitch(mergeSort(start), mergeSort(last));
};

const stitch = (start, last) => {
  const results = [];
  while (start.length && last.length) {
    if (start[0] < last[0]) {
      results.push(start.shift());
    } else {
      results.push(last.shift());
    }
  }

  return [...results, ...start, ...last];
};
